- Description: qeth: incomplete version string of qeth module
  Symptom:     An incomplete version string is diplayed when loading
               the qeth module.
  Problem:     Not all files belonging to qeth have a revision string
               defined.
  Solution:    Add missing revision string definitions.
  Problem-ID:  --
  Archs:       s390-31, s390-64

--- linux-2.5/drivers/s390/net/qeth_main.c	27 Apr 2004 15:30:15 -0000	1.77.2.6
+++ linux-2.5/drivers/s390/net/qeth_main.c	14 May 2004 07:27:37 -0000	1.77.2.7
@@ -1,6 +1,6 @@
 /*
  * 
- * linux/drivers/s390/net/qeth_main.c ($Revision: 1.77.2.6 $)
+ * linux/drivers/s390/net/qeth_main.c ($Revision: 1.77.2.7 $)
  *
  * Linux on zSeries OSA Express and HiperSockets support
  *
@@ -12,7 +12,7 @@
  *			  Frank Pavlic (pavlic@de.ibm.com) and
  *		 	  Thomas Spatzier <tspat@de.ibm.com>
  *
- *    $Revision: 1.77.2.6 $	 $Date: 2004/04/27 15:30:15 $
+ *    $Revision: 1.77.2.7 $	 $Date: 2004/05/14 07:27:37 $
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -78,10 +78,9 @@
 #include "qeth_mpc.h"
 #include "qeth_fs.h"
 
-#define VERSION_QETH_C "$Revision: 1.77.2.6 $"
-static const char *version = "qeth S/390 OSA-Express driver ("
-	VERSION_QETH_C "/" VERSION_QETH_H "/" VERSION_QETH_MPC_H
-	QETH_VERSION_IPV6 QETH_VERSION_VLAN ")";
+#define VERSION_QETH_C "$Revision: 1.77.2.7 $"
+static const char *version = "qeth S/390 OSA-Express driver";
+
 /**
  * Debug Facility Stuff
  */
@@ -6777,7 +6776,12 @@
 	int rc=0;
 	
 	qeth_eyecatcher();
-	printk(KERN_INFO "qeth: loading %s\n",version);
+	PRINT_INFO("loading %s (%s/%s/%s/%s/%s/%s/%s %s %s)\n",
+		   version, VERSION_QETH_C, VERSION_QETH_H,
+		   VERSION_QETH_MPC_H, VERSION_QETH_MPC_C,
+		   VERSION_QETH_FS_H, VERSION_QETH_PROC_C,
+		   VERSION_QETH_SYS_C, QETH_VERSION_IPV6,
+		   QETH_VERSION_VLAN);
 
 	INIT_LIST_HEAD(&qeth_card_list.list);
 	rwlock_init(&qeth_card_list.rwlock);

--- linux-2.5/drivers/s390/net/qeth_fs.h	1 Apr 2004 06:53:03 -0000	1.7
+++ linux-2.5/drivers/s390/net/qeth_fs.h	14 May 2004 07:27:37 -0000	1.7.2.1
@@ -12,6 +12,11 @@
 #ifndef __QETH_FS_H__
 #define __QETH_FS_H__
 
+#define VERSION_QETH_FS_H "$Revision: 1.7.2.1 $"
+
+extern const char *VERSION_QETH_PROC_C;
+extern const char *VERSION_QETH_SYS_C;
+
 #ifdef CONFIG_PROC_FS
 extern int
 qeth_create_procfs_entries(void);

--- linux-2.5/drivers/s390/net/qeth_sys.c	27 Apr 2004 15:30:16 -0000	1.19.2.3
+++ linux-2.5/drivers/s390/net/qeth_sys.c	14 May 2004 07:27:38 -0000	1.19.2.4
@@ -1,6 +1,6 @@
 /*
  *
- * linux/drivers/s390/net/qeth_sys.c ($Revision: 1.19.2.3 $)
+ * linux/drivers/s390/net/qeth_sys.c ($Revision: 1.19.2.4 $)
  *
  * Linux on zSeries OSA Express and HiperSockets support
  * This file contains code related to sysfs.
@@ -19,6 +19,8 @@
 #include "qeth.h"
 #include "qeth_mpc.h"
 #include "qeth_fs.h"
+
+const char *VERSION_QETH_SYS_C = "$Revision: 1.19.2.4 $";
 
 /*****************************************************************************/
 /*                                                                           */

--- linux-2.5/drivers/s390/net/qeth_mpc.h	23 Mar 2004 17:14:36 -0000	1.27
+++ linux-2.5/drivers/s390/net/qeth_mpc.h	14 May 2004 07:27:38 -0000	1.27.2.1
@@ -14,7 +14,9 @@
 
 #include <asm/qeth.h>
 
-#define VERSION_QETH_MPC_H "$Revision: 1.27 $"
+#define VERSION_QETH_MPC_H "$Revision: 1.27.2.1 $"
+
+extern const char *VERSION_QETH_MPC_C;
 
 #define IPA_PDU_HEADER_SIZE	0x40
 #define QETH_IPA_PDU_LEN_TOTAL(buffer) (buffer+0x0e)

